chore: update golangci-lint configuration

jawnsy opened the pull request.

f0ssel reviewed (approved):
not sure what all the diff really is but I trust

jawnsy commented:
@f0ssel yea it was a lot of changes, I tried to format things to better match the monorepo settings, and also enabled more linters as long as they were passing. I removed exclusions which don't seem to be needed anymore (I guess we fixed code in the meantime)
